File: 908. Smallest Range I.rb
#* 908. Smallest Range I
nums = [0,10]
k = 2
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def smallest_range_i(nums, k)
  [nums.max - nums.min - 2 * k, 0].max
end
p smallest_range_i(nums, k)

def smallest_range_i(nums, k)
  min, max = nums.minmax
  [max - min - 2 * k, 0].max
end
p smallest_range_i(nums, k)

File: 908. Smallest Range I.js
// 908. Smallest Range I
nums = [0,10]
k = 2
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var smallestRangeI = function(nums, k) {
  const maxNum = Math.max(...nums);
  const minNum = Math.min(...nums);
  return Math.max(maxNum - minNum - 2 * k, 0);
}
console.log(smallestRangeI(nums,k))

var smallestRangeI = function(nums, k) {
  return Math.max(Math.max(...nums) - Math.min(...nums) - 2 * k, 0);
}
console.log(smallestRangeI(nums,k))

